source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -1068,4 +1068,12 @@
     #define ipconfigPROCESS_CUSTOM_ETHERNET_FRAMES    0
 #endif
 
+/* This option adds the possibility to have a user-ID attached to a socket.
+ * The type of this ID is a void *.  Both UDP and TCP sockets have
+ * this ID. It has a default value of NULL.
+ */
+#ifndef ipconfigUSE_SetSocketID
```

Comment:
Do we really need one more config?

Reply:
Yes why not add it unconditionally. Linkers are smart enough to eliminate dead code. Only MISRA won't like it.
So I also vote to leave out the macro.

Reply:
> Yes why not add it unconditionally. Linkers are smart enough to eliminate dead code. Only MISRA won't like it. So I also vote to leave out the macro.

There is only one thing: all sockets will grown 4 of 8 bytes in size:
~~~
#if ( ipconfigUSE_SetSocketID != 0 )
    void * pvSocketID;
#endif
~~~
when the ID pointer is declared unconditionally.

Reply:
Yes, that is the cost of cleanliness in the code I suppose.
We already have far too many configuration options in my opinion. And the list just keeps growing.
I don't mind adding configs for things which affect the performance or certain characteristics (like timeouts etc) of the stack. But this seems like a bit of an overkill to me.
What do you think @htibosch?